### DISCUSSION TITLE
resurrection-JyersUI

### DISCUSSION THREAD

**Discussion by Jyers** [Jyers/Marlin#1891] [General Discussion]
Just a quick update so everyone knows what's going on.
I've now begun work with @tititopher68-dev on bringing his features and updates to the firmware.
The ressurection-JyersUI branch will serve as a dev branch while we test and merge all of his work into the main source.
If you want to test feel free to compile from that branch and report on any issues you find. It should be set up in a few days.
There might not be any releases for a bit while we work on the branch but once it's ready to be merged we should be ready for a long overdue new release.

**Comment by SikVac** [comment]
Fix:
resurrection-JyersUI 
(https://github.com/Jyers/Marlin/tree/resurrection-JyersUI)

To complete the compilation I had to change in the file:
Marlin/configuration.h
Old line:
2866  // #define FAN_SOFT_PWM
New line:
2866  #define FAN_SOFT_PWM

Configuration:  Ender 3V2 v422 RET6 UBL-BLTouch-10x10

**Reply by Jyers** [reply]
Yes, this is a leftover issue from the last merge I did upstream. All such things should get ironed out as I begin merging in the new code

**Comment by VashBaldeus** [comment]
> Just a quick update so everyone knows what's going on. I've now begun work with @tititopher68-dev on bringing his features and updates to the firmware. The ressurection-JyersUI branch will serve as a dev branch while we test and merge all of his work into the main source. If you want to test feel free to compile from that branch and report on any issues you find. It should be set up in a few days. There might not be any releases for a bit while we work on the branch but once it's ready to be merged we should be ready for a long overdue new release.

Great to see you guys back at work, still on 1.3.5 due tot he fact 2.0 ain't as stable as it was fresh back then.

**Comment by colonius-007** [comment]
Prima, dass es weitergeht und vielleicht bald Updates kommen. Wobei ich leider nicht selbst kompilieren kann. Ich bin froh mit meinen 71 Jahren überhaupt in den 3d Druck den Einstieg gefunden zu haben. Ich freue mich auf neue bin Dateien. Liebe Grüße aus dem schönen Köln in Germany
Kalle

**Comment by ledcomp** [comment]
Just curious if there are any updates on this project?

**Reply by colonius-007** [reply]
Thanks very mutch and Greatings from Germany

**Reply by ledcomp** [reply]
That's great progress is being made. Looking forward to a new release.

Thanks again

**Comment by DeNohvo** [comment]
Man I held off installing 2.0.1 do to the noted bugs for months, decided today to reset and install the UBL 2.0.1 and then saw this. 1.3.5 was running good but at the time didn't flash UBL version.

Ohh well, great to see development is still in progress.

**Comment by azer9999** [comment]
Hello,
thanks for your firmware, really great
i have a ender 3v2 with bl touch
could you make a version for the extended z axis too? i have no knowledge in compiling

do you have a version for the cr10v2?
thanks

**Reply by Tapap84** [reply]
it seems hard at first but compiling is not bad at all after 1-2 times. making adjustments for larger beds or z is just changing the number in one place, im working on a custom ender that's 500x500x400 and has a fixed y axis and so far all ive had to change was define bed size and reverse a pin out

**Comment by PapsDizzy** [comment]
any news about an new release?

**Reply by Tapap84** [reply]
(deleted my comment as I cant read)

Ill try your work tonight. Should I be watching your page for the updates?

This is what I have saved, I think you may have posted this but I dont remember.


Download VS Code https://code.visualstudio.com/download
Next install Platform IO extension and Auto Build Marlin extension to complete.
https://www.youtube.com/watch?v=eq_ygvHF29I 2:29 time start to 7:30 time stop, for pertinent info.
Next :
Step 1: Go to the resources section on Git :
https://github.com/Jyers/Marlin/archive/refs/heads/Ender_3_V2_JyersUI.zip
Step 2: Download the Source Code Zip.
Step 3: Unzip the zipped file.
Step 4: Launch VSCode and open the unzipped file via PlatformIO's Open Project.
Step 5: Since I am needing the configuration files for the BL Touch and not wanting it with high speed, I would access the following file location. Configuration Files => E3V2 Templates => BLTouch-3x3-HighSpeed.
Step 6: In the BLTouch-3x3-HighSpeed folder select both configuration.h & configuration_adv.h files, right click them and select copy.
Step 7: Go to the Marlin section, right click and paste to replace the existing configuration.h & configuration_adv.h h files with the ones from the BLTouch-3x3-HighSpeed folder.
Step 8: Make required changes.
In configuration.h :
For 5x5, line 1593 :
#define GRID_MAX_POINTS_X 5
For the Bed :
#define X_BED_SIZE 310
#define Y_BED_SIZE 310
#define Z_MAX_POS 400
Step 9: Save everything
Step 10: Compile Build with checkmark icon.
Step 11: Grab the newly created .bin file and put on empty SD card (FAT32 - 4096 bytes formatted).
Step 12: Plug SD card into printer which should be powered off.
Step 13: Power printer on and printer should flash to the new firmware (10 to 20s).
Once printer is fully flashed and at the menu screen, you can access the printers info section and see the updated firmware there.
Step 14: Don't forget this Golden rule :
After every flash, do a Reset to Default following by a Store Settings following by a Restore Settings to initialize correctly the EEprom.

**Reply by PapsDizzy** [reply]
ok i have 4.2.2 board with STM32F103RET6 chip so what i have to chose in platform io? is that correct default_envs = STM32F103RE_creality?

**Reply by Nbrooks639** [reply]
open project in platform, which will be this firmware file, (unzipped), edit and compile the config files for which ever leveling option you choose for your machine. once you read the the code and be careful what you uncomment because once you do that then theres more steps to back it up. then click the little check mark on the bottom to build. if its a success go up to pio right click reveal in file explorer and copy the bin. there is tons of videos on this stuff.
 not sure if you can use auto build for this. thats the hard way anyways.

**Reply by PapsDizzy** [reply]
so ok i have done the compile all looking and feeling good yet.

**Comment by Nbrooks639** [comment]
any way you could make the led menu accessible for those using neopixels. if you uncomment it......it does nothing.

**Reply by Nbrooks639** [reply]
actually you are wrong. i have them on 3 printers of mine. i only used the board for a signal pin, i have all the info you need i just dont know how to make the sub menu. 2 of my printers have the menu option in that firmware. i use 15 pixels, get power from psu with 5v step down. they work perfect. first board photo is for the s1 
![20220408_135231](https://user-images.githubusercontent.com/100983669/162623341-333894a3-e1f5-44e0-b263-c5a7550efaad.jpg)
in the photo. s1, s1, v2
![photo_2022-03-27_00-04-40](https://user-images.githubusercontent.com/100983669/162623372-95018235-a305-48c3-a0e5-244b6ddd0fe4.jpg)

https://user-images.githubusercontent.com/100983669/162623383-dea7e596-4810-4b39-8cc3-aca0184babef.mp4

![photo_2022-03-31_01-31-31](https://user-images.githubusercontent.com/100983669/162623494-b5d66758-e7a0-4387-8c60-4ddb9a86fb16.jpg)
![photo_2022-03-31_01-31-23](https://user-images.githubusercontent.com/100983669/162623495-4c0721c3-c3a8-49e2-9a9a-4c124f103d8a.jpg)
![photo_2022-04-01_15-24-39](https://user-images.githubusercontent.com/100983669/162623496-b9fd604c-4f0f-4879-8d4f-40e14454442b.jpg)


![image](https://user-images.githubusercontent.com/100983669/162623660-5ebec8db-aa0b-4355-a4f7-af701693ce0c.png)
 well here you go. all you gotta  due is activate make the sub menu, i can do the rest

**Reply by Nbrooks639** [reply]
ender 3 s1 board is the top picture, pa 13 and 14 are there on the f1 board, the marlin firmware even has this option available kinda weird. the bottom picture is for the 4.x.x boards pa 7 would be the best pin to use for signal z endstop. kind of crazy this fw is based off marlin but you wont allow full access. how would he be hiding part of it. also i noticed you mention his name in your code. man where is mr. myers

**Comment by bsinger99** [comment]
I have an Ender 3 v2 that has been upgraded with a 4.2.7 mainboard, Biqu Hermit Crab & H2 extruder, and a CR-Touch.  I'm a newbie, so if the solution is obvious, please forgive me. My current issue is with Z axis homing.  When Auto-Homing All, the gantry goes up a little, and the CR-Touch pin drops, but it never reaches the bed. I've tried disabling the steppers and manually moving the gantry down. On the homing retry, the printer either displays a 'kill' message after dropping the CR-Touch probe pin or the gantry moves up with the CR-Touch pin dropping out of range of the bed (with a 'stopped' message). Moving the gantry up and down with the Move menu works, so I'm thinking it's a firmware or wiring issue. I've tried changing the Z_MIN_PROBE_ENDSTOP_INVERTING value to true, but the firmware wouldn't compile & told me it had to be set to false. I checked the status of the end stops with an M119 (before doing any homing). The X & Y_MIN are OPEN and the Z_MIN, Z_PROBE, and Filament are TRIGGERED - which I'm guessing is a symptom of the problem. I've tried several combinations of Z_MIN_ENDSTOP_INVERTING, Z_MIN_PROBE_ENDSTOP_INVERTING, and Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN, but can't get past the firmware compilation requirement that Z_MIN_PROBE_ENDSTOP_INVERTING be set to false.
